samtranslator/model/eventsources/push.py

```diff
@@ -788,6 +788,35 @@ def _add_swagger_integration(self, api, function, intrinsics_resolver):
                     path=self.Path, method_name=self.Method, request_model=self.RequestModel
                 )
 
+                validate_body = self.RequestModel.get("ValidateBody")
+                validate_request = self.RequestModel.get("ValidateRequest")
+
+                # Checking if any of the fields are defined as it can be false we are checking if the field are not None
+                if not isinstance(validate_body, type(None)) or not isinstance(validate_request, type(None)):
+
+                    # as we are setting two different fields we are here setting as default False
+                    # In case one of them are not defined
+                    validate_body = False if type(validate_body) is None else validate_body
+                    validate_request = False if type(validate_request) is None else validate_request
+
+                    # If not type None but any other type it should explicitly invalidate the Spec
+                    # Those fields should be only a boolean
+                    if not isinstance(validate_body, bool) or not isinstance(validate_request, bool):
```

Comment:
I think this will fail if `validate_body` or `validate_request` is set to a more complex intrinsic like an If. I made a comment in the tests about this, so that should prove what happens one way or another.

Reply:
I added an If case, same behavior would be expected as setting an intrinsic definition.